src/ImmutableX.ts

```diff
@@ -0,0 +1,842 @@
+} from './api';
+import { formatError } from './utils/formatError';
+import { ImmutableXConfiguration } from './config';
+
+/**
+ * The main entry point for the Core SDK
+ */
+export class ImmutableX {
+  private depositsApi: DepositsApi;
+  private mintsApi: MintsApi;
+  private ordersApi: OrdersApi;
+  private tokensApi: TokensApi;
+  private tradesApi: TradesApi;
+  private transfersApi: TransfersApi;
+  private usersApi: UsersApi;
+  private withdrawalsApi: WithdrawalsApi;
+  private balanceApi: BalancesApi;
+  private assetApi: AssetsApi;
+  private collectionApi: CollectionsApi;
+  private metadataApi: MetadataApi;
+  private metadataRefreshesApi: MetadataRefreshesApi;
+  private projectsApi: ProjectsApi;
+  private workflows: Workflows;
+
+  constructor(config: ImmutableXConfiguration) {
+    this.depositsApi = new DepositsApi(config.apiConfiguration);
+    this.mintsApi = new MintsApi(config.apiConfiguration);
+    this.ordersApi = new OrdersApi(config.apiConfiguration);
+    this.tokensApi = new TokensApi(config.apiConfiguration);
+    this.tradesApi = new TradesApi(config.apiConfiguration);
+    this.transfersApi = new TransfersApi(config.apiConfiguration);
+    this.usersApi = new UsersApi(config.apiConfiguration);
+    this.withdrawalsApi = new WithdrawalsApi(config.apiConfiguration);
+    this.balanceApi = new BalancesApi(config.apiConfiguration);
+    this.assetApi = new AssetsApi(config.apiConfiguration);
+    this.collectionApi = new CollectionsApi(config.apiConfiguration);
+    this.metadataApi = new MetadataApi(config.apiConfiguration);
+    this.metadataRefreshesApi = new MetadataRefreshesApi(
+      config.apiConfiguration,
+    );
+    this.projectsApi = new ProjectsApi(config.apiConfiguration);
+    this.workflows = new Workflows(config);
+  }
+
+  /**
+   * Deposit based on a token type. For unregistered Users, the Deposit will be combined with a registration in order to register the User first
+   * @param ethSigner - the L1 signer
+   * @param deposit - the token type amount in its corresponding unit
+   * @returns a promise that resolves with the resulting transaction
+   * @throws {@link index.IMXError}
+   */
+  public deposit(ethSigner: EthSigner, deposit: TokenAmount) {
+    return this.workflows.deposit(ethSigner, deposit).catch(err => {
+      throw formatError(err);
+    });
+  }
+
+  /**
+   * Get details of a Deposit with the given ID
+   * @param request - the request object containing the parameters to be provided in the API request
+   * @returns a promise that resolves with the requested Deposit
+   * @throws {@link index.IMXError}
+   */
+  public getDeposit(request: DepositsApiGetDepositRequest) {
+    return this.depositsApi
+      .getDeposit(request)
+      .then(res => res.data)
+      .catch(err => {
+        throw formatError(err);
+      });
+  }
+
+  /**
+   * Get a list of Deposits
+   * @param request - the request object containing the parameters to be provided in the API request
```

Comment:
`request` is optional? Should we specify this? (not blocking)

Reply:
typedoc already marks it as optional but I will update the description from `- the request object` to `- optional request object`

![image](https://user-images.githubusercontent.com/5828814/193954470-4e27de6d-6a31-4ae7-a3af-262528b38138.png)

Reply:
that's nice, how about for intellisense?

Reply:
[05dab29](https://github.com/immutable/imx-core-sdk/pull/185/commits/05dab29a61dda6416ec0080c8c45a19130e36a68)